cmd/system/caddy.go

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) arkade author(s) 2024. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package system
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/alexellis/arkade/pkg/archive"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	"github.com/spf13/cobra"
+)
+
+func MakeInstallCaddyServer() *cobra.Command {
+	command := &cobra.Command{
+		Use:   "caddy",
+		Short: "Install Caddy Server",
+		Long:  `Install Caddy Server which is an extensible server platform that uses TLS by default`,
+		Example: `  arkade system install caddy
+  arkade system install caddy --version <version>`,
+		SilenceUsage: true,
+	}
+
+	command.Flags().StringP("version", "v", "", "The version or leave blank to determine the latest available version")
+	command.Flags().String("path", "/usr/local/bin", "Installation path, where a caddy subfolder will be created")
```

Comment:
It's not the usual pattern, but Caddy installs to /usr/bin/

Have a look at the faasd script to see what else may be missing, including adding users etc.

For adding users, we will need to detect the OS from /etc/os-release and reject the command if it's not debian or ubuntu.

In the future, if someone needs this for Alpine or Fedora, they can add a patch to run useradd with the correct syntax for their distro.

There are also some directories created and chmod is run.